### PR TITLE
feat: Add Top-N Tutorial

### PR DESCRIPTION
**Pull request by bbejeck**
[asana](https://app.asana.com/0/1201951209538300/1205643631666335/f)

### CODE REVIEW

**Comment by bbejeck** [comment]
@davetroiano - the images from cloud are a little janky - I couldn't expand the result window in cloud to get rid of scrolling so I tried three screenshots - let me know if you think if just one is good enough.

File: top-N/flinksql/README.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
+# Get top (or bottom) 'N' events with Flink SQL
+
+Suppose you some service or product, and you want to see the top uses from an eventstream.  For example, consider work for a video streaming service like NetFlix or Hulu.  You want to see the top five movies by max number of views in realtime.  To do this ranking, you can use a [Top-N query](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/topn/).
```
Comment (davetroiano) [inline top-N/flinksql/README.md:3]: missing word "Suppose you ____ some service". "Suppose you monitor ..."?

Comment (davetroiano) [inline top-N/flinksql/README.md:3]:
```suggestion
Suppose you some service or product, and you want to see the top uses from an event stream.  For example, consider working as an analyst for a video streaming service like Netflix or Hulu.  You want to see the top three movies by max number of views in real time.  To do this ranking, you can use a [Top-N query](https://nightlies.apache.org/flink/flink-docs-release-1.19/docs/dev/table/sql/queries/topn/). 
```

File: top-N/flinksql/README.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    id INT,
+    title STRING,
+    genre STRING,
+    num_views BIGINT
+);
+```
+
+## Compute the Top-N views
+
+Given the `movie_views` table definition above, we can retrieve the top three movies by genre that have max views in realtime.
```
Comment (davetroiano) [inline top-N/flinksql/README.md:20]:
```suggestion
Given the `movie_views` table definition above, we can retrieve the top three movies by genre that have max views in real time.
```

File: top-N/flinksql/README.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+SELECT title, genre, num_views, category_rank
+FROM (
+       SELECT *,
+              ROW_NUMBER() OVER (PARTITION BY genre ORDER BY num_views DESC) as category_rank
+       FROM movie_views
+     )
+WHERE category_rank <= 3;
+```
+
+The subquery here is doing the heavy lifting, so let's take a detailed look at it.  The subquery orders the movies by the number views (descending) and assigns a unique number to each row.  This process makes it possible to rank movies where the row number is less than or equal to five. Let’s discuss the critical parts of the subquery:
```
Comment (davetroiano) [inline top-N/flinksql/README.md:32]:
```suggestion
The subquery here is doing the heavy lifting, so let's take a detailed look at it.  The subquery orders the movies by the number of views (descending) and assigns a unique number to each row.  This process makes it possible to rank movies where the row number is less than or equal to three. Let’s discuss the critical parts of the subquery:
```

File: top-N/flinksql/README.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  * A Flink compute pool created in Confluent Cloud. Follow [this](https://docs.confluent.io/cloud/current/flink/get-started/quick-start-cloud-console.html) quick start to create one.
+
+  #### Run the commands
+
+  In the Confluent Cloud Console, navigate to your environment and then click the `Open SQL Workspace` button for the compute
+  pool that you have created.
+
+  Select the default catalog (Confluent Cloud environment) and database (Kafka cluster) to use with the dropdowns at the top right.
+
+  Finally, run following SQL statements to create the `movie_sales` table, populate it with test data, and run the aggregating min/max query.
```
Comment (davetroiano) [inline top-N/flinksql/README.md:187]:
```suggestion
  Finally, run following SQL statements to create the `movie_views` table, populate it with test data, and run the Top-N query.
```

**Review by davetroiano — APPROVED** [review]
LGTM!

**Pull request merged**